lib/collection/src/collection/resharding.rs

```diff
@@ -210,6 +210,10 @@ impl Collection {
             },
         }
 
+        shard_holder
+            .abort_resharding(resharding_key.clone(), force)
+            .await?;
+
```

Comment:
Additionally: abort before persisting the shard number on disk.

This way we don't persist wrong state if aborting itself fails.